state/protocol/badger/epoch.go

```diff
@@ -112,9 +126,10 @@ func (es *CommittedEpoch) DKG() (protocol.DKG, error) {
 	return &DKG{commitEvent: es.commitEvent}, nil
 }
 
-func NewCommittedEpoch(setupEvent *flow.EpochSetup, commitEvent *flow.EpochCommit) *CommittedEpoch {
+func (q *EpochQuery) NewCommittedEpoch(setupEvent *flow.EpochSetup, commitEvent *flow.EpochCommit) *CommittedEpoch {
```

Comment:
can we make this method _not_ exported?

Reply:
Sorry, my comment was probably not very helpful. I explained in [more detail below](https://github.com/onflow/flow-go/pull/157#pullrequestreview-538021201)
> **(2) The implementation of protocol.Epoch is internally based on a Snapshot for a specific block**

Reply:
updated per our conversation to cache first view in the epoch status